docs/documents/span.dox

```diff
@@ -2,9 +2,36 @@
  *
  * @copyright Copyright Tristan Brindle 2018.
  *
- * @brief A C++ 11 implementation of the non-owning C++20 std::span type.
+ * @brief A C++ 11 implementation of the non-owning C++20 \ref std::span type.
  *
- * A span represents a non-owning view into a contiguous block of memory of a known size.
+ * A \ref span represents a non-owning view into a contiguous block of memory of a known size.
+ * This view can be mutable or const, depending on the type.
  *
- * This version contains an additional constructor for `const char *`-style C strings, allowing the span to mimic C++17 std::string_view.
+ * A simple mental model is an object that holds a pointer and size, so it knows how many elements it is allowed to access after the initial pointer.
+ * You can imagine a `span<double>` to look like this:
+ *
+ * @code cpp
+ * class span {
+ *   double* begin;
+ *   size_t size;
+ * }
+ * @endcode
+ *
+ * This type can also be read-only aka `const`. You can imagine a `span<const double>` to look like this:
+ *
+ * @code cpp
+ * class span {
+ *   double const* begin;
+ *   size_t size;
```

Comment:
it would also be `size_t const size` right?

Reply:
I had a similar conversation with @IshaanDesai lately.
While the viewed data cannot be changed in this example, the view itself can.
The pointed-to data is constant, not the pointer itself.

For example, one can still move the start of it.